src/charm.py

```diff
@@ -172,6 +169,9 @@ def _create_memory_storage(self, path: Path, size: int) -> None:
         Raises:
             RunnerError: Unable to setup storage for runner.
         """
+        if size <= 0:
+            return
+
```

Comment:
May I ask if this should be an exception? Or maybe a validation error upstream since size being 0 or less seems like not a sensible value. :D

Reply:
If there is 0 runners, the allocation size will be 0.
But there should be not case where size is under 0.